packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-cognito-app-client-secret.ts

```diff
@@ -0,0 +1,51 @@
+import { $TSContext, AmplifyCategories, stateManager } from '@aws-amplify/amplify-cli-core';
+import { getAuthResourceName } from '../../../utils/getAuthResourceName';
+import { AuthInputState } from '../auth-inputs-manager/auth-input-state';
+import { MetaOutput } from '../import/types';
+import { getAppClientSecretViaSdk } from './get-app-client-secret-sdk';
+import { projectHasAuth } from './project-has-auth';
+
+/**
+ *
+ * returns app client secret if user pool exists
+ */
+const generateAppClientSecret = async (
+  context: $TSContext,
+  authResourceName: string,
+  clientId: string,
+  userPoolId: string,
+): Promise<string | undefined> => {
+  const authState = new AuthInputState(context, authResourceName);
+  if (authState.cliInputFileExists()) {
+    const { cognitoConfig } = authState.getCLIInputPayload();
+    if (cognitoConfig.userpoolClientGenerateSecret) {
+      const appClientSecret = await getAppClientSecretViaSdk(context, userPoolId, clientId);
+      console.log(appClientSecret);
+      return appClientSecret;
+    }
+  }
+};
+
+/**
+ *
+ * updates app client secret if user pool exists and userpoolClientGenerateSecret is set to true
+ */
+export const updatesAppClientSecret = async (context: $TSContext): Promise<void> => {
```

Comment:
if this function is responsible for both generation and update then it should be renamed to reflect that.
I.e. `updateAppClientWithGeneratedSecret` ? otherwise logic should be split up and caller should take resposibility of generating and then updating the secret.